src/DurableTask.AzureStorage/Tracking/AzureTableTrackingStore.cs

```
@@ -1211,9 +1245,10 @@ static string GetBlobName(DynamicTableEntity entity, string property)
             {
                 for (int i = tableResultList.Count - 1; i >= 0; i--)
                 {
-                    if (((DynamicTableEntity)tableResultList[i].Result).RowKey == SentinelRowKey)
+                    DynamicTableEntity resultEntity = (DynamicTableEntity)tableResultList[i].Result;
```

I realize that this is just a readability fix, so this code has been working already for a long time, but how strong is our guarantee that `tableResultList[i]` is actually a `DynamicTableEntity`? It might be safer to use the `as` operator here and use the `?.` operator in the line below, but if we have strong enough guarantees, then we don't necessarily need to add the extra null checking overhead.

---

I don't know how strong the guarantee is, but I would rather it fail with an `InvalidCastException` than risk a hard-to-find bug related to missing eTags. In fact, now that I'm looking at this code more closely, I would like to remove the surrounding `if (tableResultList != null)` check. I figure if the assumption is ever not valid, we'll find out about it sooner rather than later.